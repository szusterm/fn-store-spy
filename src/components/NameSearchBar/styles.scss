@import '../../scss/shadow';
@import '../../scss/transition';
@import '../../scss/whiteBackground';

.name-search-bar {
	$inputHeight: 50px;

	width: 100%;
	padding: 0 15px;

	&--input {
		width: 100%;
		height: $inputHeight;
		padding: 0 $inputHeight/2;

		border-width: 0;
		border-radius: $inputHeight/2;

		@include whiteBackground(0.6);
		@include shadow(sm);

		@include transition(sh);

		&:hover, &:focus {
			@include whiteBackground(0.9);
			@include shadow(md);
		}
	}
}