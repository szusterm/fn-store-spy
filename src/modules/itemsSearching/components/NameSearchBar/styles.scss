@import '~scss/_shadow';
@import '~scss/_transition';
@import '~scss/_opacity-colors';
@import '~scss/_input';

.name-search-bar {
	$inputHeight: 50px;

	width: 100%;
	padding: 0 15px;

	&--input {
		@include default-input();
		cursor: auto;
		width: 100%;
		height: $inputHeight;
		padding: 0 $inputHeight/2;

		border-width: 0;
		border-radius: $inputHeight/2;

		@include white-background(0.95);
		@include shadow(sm);

		@include transition(sh);

		&:hover, &:focus {
			@include white-background(1);
			@include shadow(md);
		}
	}
}