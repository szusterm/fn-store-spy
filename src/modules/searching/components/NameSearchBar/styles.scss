@import '../../../../scss/shadow';
@import '../../../../scss/transition';
@import '../../../../scss/opacity-colors';
@import '../../../../scss/input-reset';

.name-search-bar {
	$inputHeight: 50px;

	width: 100%;
	padding: 0 15px;

	&--input {
		@include input-reset();
		width: 100%;
		height: $inputHeight;
		padding: 0 $inputHeight/2;

		border-width: 0;
		border-radius: $inputHeight/2;

		@include white-background(0.7);
		@include shadow(sm);

		@include transition(sh);

		&:hover, &:focus {
			@include white-background(0.9);
			@include shadow(md);
		}
	}
}