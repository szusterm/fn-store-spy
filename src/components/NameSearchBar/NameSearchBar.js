import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {setNameFilter} from '../../redux/actions';

import './styles.scss';

export class NameSearchBar extends Component {
	render() {
		return (
			<div className={'name-search-bar'}>
				<input
					className={'name-search-bar--input'}
					placeholder={'Search by name'}
				/>
			</div>
		);
	}
}

NameSearchBar.propTypes = {
	name: PropTypes.string, //redux
	setNameFilter: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {name} = state.searching.filters;
	return {name};
};

const mapDispatchToProps = {setNameFilter};

export default connect(mapStateToProps, mapDispatchToProps)(NameSearchBar);