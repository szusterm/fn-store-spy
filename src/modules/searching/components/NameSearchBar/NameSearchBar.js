import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {setNameFilter} from '../../../../redux/actions';

import './styles.scss';

export class NameSearchBar extends Component {
	constructor() {
		super();

		this.updateStore = this.updateStore.bind(this);
	}

	updateStore(event) {
		const {value: name} = event.target;
		const {minNameFilterLength} = this.props;

		const nameToSave = (name.length >= minNameFilterLength) ? name : '';
		this.props.setNameFilter(nameToSave);
		this.props.onChange();
	}

	render() {
		const {maxNameFilterLength} = this.props;

		return (
			<div className={'name-search-bar'}>
				<input
					className={'name-search-bar--input'}
					placeholder={'Search by name'}
					maxLength={maxNameFilterLength}
					onChange={this.updateStore}
				/>
			</div>
		);
	}
}

NameSearchBar.propTypes = {
	onChange: PropTypes.func,
	maxNameFilterLength: PropTypes.number, //redux
	minNameFilterLength: PropTypes.number, //redux
	name: PropTypes.string, //redux
	setNameFilter: PropTypes.func //redux
};

const mapStateToProps = (state) => {
	const {name} = state.searching.filters;
	const {minNameFilterLength, maxNameFilterLength} = state.config;

	return {
		name,
		minNameFilterLength,
		maxNameFilterLength
	};
};

const mapDispatchToProps = {setNameFilter};

export default connect(mapStateToProps, mapDispatchToProps)(NameSearchBar);