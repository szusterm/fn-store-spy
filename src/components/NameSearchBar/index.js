import NameSearchBar from './NameSearchBar';

export default NameSearchBar;