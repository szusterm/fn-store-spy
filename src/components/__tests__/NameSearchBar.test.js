import React from 'react';
import {shallow} from 'enzyme';

import {NameSearchBar} from '../NameSearchBar/NameSearchBar';

const setup = (propOverrides) => {
	const props = Object.assign({}, propOverrides);

	const wrapper = shallow(<NameSearchBar {...props}/>);

	return {props, wrapper};
};

describe('NameSearchBar Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});
});