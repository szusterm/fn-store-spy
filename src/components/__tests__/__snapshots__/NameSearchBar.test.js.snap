// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameSearchBar Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NameSearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="name-search-bar--input"
        onChange={[Function]}
        placeholder="Search by name"
      />,
      "className": "name-search-bar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "name-search-bar--input",
        "maxLength": undefined,
        "onChange": [Function],
        "placeholder": "Search by name",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="name-search-bar--input"
          onChange={[Function]}
          placeholder="Search by name"
        />,
        "className": "name-search-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "name-search-bar--input",
          "maxLength": undefined,
          "onChange": [Function],
          "placeholder": "Search by name",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
